backend/common/compute/estimate_distribution.py

```diff
@@ -5,27 +5,46 @@
 from backend.common.constants import XApproximateDistribution
 
 
-@numba.njit(fastmath=True, error_model="numpy", nogil=True)
-def min_max(arr):
+@numba.njit(error_model="numpy", nogil=True)
+def min_max(arr: np.ndarray):
     """Return (min, max) values for the ndarray."""
     n = arr.size
     odd = n % 2
-    if not odd:
+    if odd:
         n -= 1
-    max_val = min_val = arr[0]
-    i = 1
+
+    # initialize to first finite value in array.  Normally,
+    # this will exit on the first value.
+    for i in range(arr.size):
```

Comment:
can you use n instead of arr.size here? 
also could you rename n to array_length?

Reply:
you can't use `n` here, as it may have been modified already and not be the actual array length (`n` is the loop unrolling limit).

I will improve variable naming.